### PR TITLE
An attempt at fixing tests on master

### PR DESCRIPTION
**Pull request by eranrund**
Lets see if this pleases CI.

### CODE REVIEW

**Comment by eranrund** [comment]
@yaahc unfortunately it appears different rust versions produce different results :(

**Comment by yaahc** [comment]
yea, it looks like we might need to add some custom post processing to strip out whitespace and `*`'s or something if we want to support block doc comments consistently across versions. I'm now kicking myself because I never setup a regular CI job to test new nightly versions so I completely missed this change going into stable :(.

**Comment by eranrund** [comment]
I pushed a dirty hack to work around the `*`s but now nightly is failing because it appears to be generating a different error message... :(

**Comment by eranrund** [comment]
Not the most elegant workaround but hopefully good enough...

File: src/attr.rs [hunk 1]
```diff
@@ -29,7 +29,14 @@ pub fn display(attrs: &[Attribute]) -> Result<Option<Display>> {
                 _ => unimplemented!(),
             };
 
-            let lit = LitStr::new(lit.value().trim(), lit.span());
+            let lit = LitStr::new(
+                &lit.value()
+                    .trim()
+                    // Deal with rustc including unnecessary parts of the doc comment :(
+                    .replace("\n     *", "\n")
```
Comment (yaahc) [inline src/attr.rs:36]: this will probably need to handle various amounts of whitespace, since the doc comment could be on a type in an inline module for example. It's probably worth adding another test case with a higher level of indentation to ensure that we correctly strip the whitespace in both cases.

**Comment by yaahc** [comment]
> Not the most elegant workaround but hopefully good enough...

yea, I'm sorry about that and thank you again for helping fix this for me :heart:

**Comment by eranrund** [comment]
No problem! Thank you for maintaining this crate :)

File: src/attr.rs (hunk 1 above)
Comment (eranrund) [reply src/attr.rs:36]: Here comes the regexs... I'll add the test case and address it when I get a chance.

Comment (eranrund) [reply src/attr.rs:36]: So this can either be solved with a regex, or with code that has some knowledge about the formatting we're trying to work around. Something along the lines of if all lines in a comment except the first one start with the same amount of spaces followed by `*`, then strip that.
Do you have a preference?

Comment (yaahc) [reply src/attr.rs:36]: I was thinking split the string with `lines()`, then use `trim()` on each str, which should hopefully give `* <the rest of each comment line>` which can then be fixed with a consistent `replace("* ", "")` or something

Comment (eranrund) [reply src/attr.rs:36]: I initially thought about doing that but I wonder what happens in the following:
```
/// Some list:
/// * a
/// * b
/// * c
```
This could be detected by the comment not starting with a `*` (from the `/**` part).

Comment (yaahc) [reply src/attr.rs:36]: This might be okay since we're already banning these kinds of doc comments with your other change

Comment (eranrund) [reply src/attr.rs:36]: Fair point :)

**Review by yaahc — APPROVED** [review]
